trinity/sync/full/chain.py

```diff
@@ -85,314 +143,194 @@ class FastChainSyncer(BaseHeaderChainSyncer):
-                                     all_headers: Tuple[BlockHeader, ...]
-                                     ) -> Dict[Tuple[Hash32, Hash32], BlockBody]:
-        """
-        Downloads and persists the block bodies for the given set of block headers.
-        Block bodies are requested from all peers in equal sized batches.
-        """
-        headers = tuple(header for header in all_headers if not _is_body_empty(header))
-        block_bodies_by_key: Dict[Tuple[Hash32, Hash32], BlockBody] = {}
-
-        while headers:
-            # split the remaining headers into equal sized batches for each peer.
-            peers = cast(Tuple[ETHPeer, ...], self.peer_pool.get_peers(target_td))
-            if not peers:
-                raise NoEligiblePeers(
-                    "No connected peers have the block bodies we need for td={0}".format(target_td)
+        try:
+            if len(non_trivial_headers) == 0:
+                self.logger.debug(
+                    "Block bodies for all %d headers were trivial, from %r..%r",
+                    len(all_headers),
+                    all_headers[0],
+                    all_headers[-1],
                 )
-            batch_size = math.ceil(len(headers) / len(peers))
-            batches = tuple(partition_all(batch_size, headers))
-
-            # issue requests to all of the peers and wait for all of them to respond.
-            requests = tuple(
-                self._get_block_bodies(peer, batch)
-                for peer, batch
-                in zip(peers, batches)
-            )
-            responses = await self.wait(asyncio.gather(
-                *requests,
-                loop=self.get_event_loop(),
-            ))
-
-            # extract the returned block body data and the headers for which we
-            # are still missing block bodies.
-            all_block_body_bundles, all_missing_headers = zip(*responses)
-
-            for (body, (tx_root, trie_data_dict), uncles_hash) in concat(all_block_body_bundles):
-                await self.wait(self.db.coro_persist_trie_data_dict(trie_data_dict))
+                completed_headers = non_trivial_headers
+            else:
+                bundles, completed_headers = await self._get_block_bodies(peer, non_trivial_headers)
+                await self._block_body_bundle_processing(bundles)
 
-            block_bodies_by_key = merge(block_bodies_by_key, {
-                (transaction_root, uncles_hash): block_body
-                for block_body, (transaction_root, trie_dict_data), uncles_hash
-                in concat(all_block_body_bundles)
-            })
-            headers = tuple(concat(all_missing_headers))
+            self._mark_body_download_complete(batch_id, completed_headers + trivial_headers)
+        except BaseP2PError:
+            self._block_body_tasks.complete(batch_id, trivial_headers)
+            self.logger.debug("Problem downloading body from peer, dropping...", exc_info=True)
```

Comment:
This blanket catch of any `p2p` exception stands out to me as potentially problematic. What do you think about adding an `INFO` level statement so that we can zero in on being more explicit about what we catch?